### PR TITLE
feat(ingest/mode): Mode improvements:

### PR DESCRIPTION
**Pull request by treff7es**
- column level lineage
- external links
- stateful ingestion support
- owner tags
- etc ...

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/tests/integration/mode/test_mode.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 FROZEN_TIME = "2021-12-07 07:00:00"
 
 JSON_RESPONSE_MAP = {
+    "https://app.mode.com/api/verify": "verify.json",
```
Comment (hsheth2) [inline metadata-ingestion/tests/integration/mode/test_mode.py:15]: is this file missing?

File: metadata-ingestion/src/datahub/ingestion/source/mode.py [hunk 2]
```diff
@@ -583,9 +669,327 @@ def _get_datasource_urn(self, platform, platform_instance, database, source_tabl
...
+                    aspect=schema_metadata,
+                ).as_workunit()
+            )
+
+        yield from self.get_upstream_lineage_for_parsed_sql(
+            query_data, parsed_query_object
+        )
+
+        operation = OperationClass(
+            operationType="UPDATE",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/mode.py:818]: let's use the model's enum here?

File: metadata-ingestion/src/datahub/ingestion/source/mode.py [hunk 3]
```diff
@@ -583,9 +669,327 @@ def _get_datasource_urn(self, platform, platform_instance, database, source_tabl
...
+                source=QuerySourceClass.SYSTEM,
+                created=AuditStamp(time=created_ts, actor=modified_actor),
+                lastModified=AuditStamp(time=modified_ts, actor=modified_actor),
+            )
+
+            yield MetadataChangeProposalWrapper(
+                entityType="query",
+                changeType=ChangeTypeClass.UPSERT,
+                entityUrn=query_instance_urn,
+                aspectName="queryProperties",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/mode.py:864]: only entityUrn and aspect are required - everything else can be inferred automatically

**Review by hsheth2 — APPROVED** [review]
LGTM

Eventually it'd be better to separate out the API interface / API client from the main source, but that can be done later

File: metadata-ingestion/tests/integration/mode/test_mode.py (hunk 1 above)
Comment (treff7es) [reply metadata-ingestion/tests/integration/mode/test_mode.py:15]: Interesting, it seems like it never was part of the test resources

Comment (treff7es) [reply metadata-ingestion/tests/integration/mode/test_mode.py:15]: I found it in the meantime :)

**Pull request merged**